File: apt3060/android-studio/end-sem/app/src/main/java/com/example/calculator/MainActivity.java
package com.example.calculator;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import androidx.appcompat.app.AppCompatActivity;
import java.util.regex.Pattern;

public class MainActivity extends AppCompatActivity {
    private EditText editTextCalculator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextCalculator = findViewById(R.id.editTextCalculator);
    }

    public void appendToCalculator(View view) {
        Button button = (Button) view;
        String buttonText = button.getText().toString();
        String currentText = editTextCalculator.getText().toString();
        editTextCalculator.setText(currentText + buttonText);
    }

    public void clearCalculator(View view) {
        editTextCalculator.setText("");
    }

    public void calculateResult(View view) {
        String expression = editTextCalculator.getText().toString().trim();

        if (expression.isEmpty()) {
            editTextCalculator.setText("Error: Empty input");
            return;
        }

        String[] operators = {"+", "-", "*", "/"};

        String operator = null;
        double num1, num2;

        for (String op : operators) {
            if (expression.contains(op)) {
                operator = op;
                break;
            }
        }

        if (operator == null) {
            editTextCalculator.setText("Error: Invalid expression");
            return;
        }

        String[] parts = expression.split(Pattern.quote(operator));

        if (parts.length != 2) {
            editTextCalculator.setText("Error: Invalid expression");
            return;
        }

        try {
            num1 = Double.parseDouble(parts[0].trim());
            num2 = Double.parseDouble(parts[1].trim());

            double result = 0;

            switch (operator) {
                case "+":
                    result = num1 + num2;
                    break;
                case "-":
                    result = num1 - num2;
                    break;
                case "*":
                    result = num1 * num2;
                    break;
                case "/":
                    if (num2 != 0) {
                        result = num1 / num2;
                    } else {
                        editTextCalculator.setText("Error: Division by zero");
                        return;
                    }
                    break;
                default:
                    editTextCalculator.setText("Error: Unsupported operator");
                    return;
            }

            editTextCalculator.setText(Double.toString(result));
        } catch (NumberFormatException e) {
            editTextCalculator.setText("Error: Invalid numbers");
        }
    }

}

File: major/apt3060/end-sem/q1/Xamarin/Calculator/MainActivity.cs
using System;
using Android.App;
using Android.OS;
using Android.Views;
using Android.Widget;

namespace CalculatorApp
{
    [Activity(Label = "MainActivity", MainLauncher = true)]
    public class MainActivity : Activity
    {
        private EditText editTextCalculator;

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_main);

            editTextCalculator = FindViewById<EditText>(Resource.Id.editTextCalculator);
        }

        public void AppendToCalculator(View view)
        {
            Button button = (Button)view;
            string buttonText = button.Text;
            string currentText = editTextCalculator.Text;
            editTextCalculator.Text = currentText + buttonText;
        }

        public void ClearCalculator(View view)
        {
            editTextCalculator.Text = "";
        }

        public void CalculateResult(View view)
        {
            string expression = editTextCalculator.Text.Trim();

            if (string.IsNullOrEmpty(expression))
            {
                editTextCalculator.Text = "Error: Empty input";
                return;
            }

            string[] operators = { "+", "-", "*", "/" };

            string operatorStr = null;
            double num1, num2;

            foreach (string op in operators)
            {
                if (expression.Contains(op))
                {
                    operatorStr = op;
                    break;
                }
            }

            if (operatorStr == null)
            {
                editTextCalculator.Text = "Error: Invalid expression";
                return;
            }

            string[] parts = expression.Split(new[] { operatorStr }, StringSplitOptions.None);

            if (parts.Length != 2)
            {
                editTextCalculator.Text = "Error: Invalid expression";
                return;
            }

            try
            {
                num1 = double.Parse(parts[0].Trim());
                num2 = double.Parse(parts[1].Trim());

                double result = 0;

                switch (operatorStr)
                {
                    case "+":
                        result = num1 + num2;
                        break;
                    case "-":
                        result = num1 - num2;
                        break;
                    case "*":
                        result = num1 * num2;
                        break;
                    case "/":
                        if (num2 != 0)
                        {
                            result = num1 / num2;
                        }
                        else
                        {
                            editTextCalculator.Text = "Error: Division by zero";
                            return;
                        }
                        break;
                    default:
                        editTextCalculator.Text = "Error: Unsupported operator";
                        return;
                }

                editTextCalculator.Text = result.ToString();
            }
            catch (FormatException)
            {
                editTextCalculator.Text = "Error: Invalid numbers";
            }
        }
    }
}
